### DISCUSSION TITLE
Can this library work with oauth

### DISCUSSION THREAD

**Discussion by tonytakumi** [NikolaGavric94/laravel-square#68] [Q&A] [answered]
Hello Nikola et all, 
Thanks for this library and the effort.
How would I go about modifying this library to take payments for another account through oAuth.
I have the permissions granted , the merchant id and access token but i don't know how to charge with these details?
Any help would be great

Thanks
Tony

**Comment by NikolaGavric94** [comment]
All you need to be able to do payments with the library are ApplicationID and Token that you can get from Square. oAuth isn't limiting this as it is used to authenticate your application users, not the square vault. As long as you have appID and token from Square you can even inject them during runtime to query against different merchants

**Reply by tonytakumi** [reply]
Thanks for the quick response. 
Could you show a little example of injecting the appID and token during runtime please?

**Comment by NikolaGavric94** [comment, marked as answer]
Based on [this](https://laravel.com/docs/8.x/configuration#accessing-configuration-values), something like this:
```
config(['nikolag.connections.square.access_token' => '<YOUR_ACCESS_TOKEN>']);
```
